go/bind/notifications.go

```diff
@@ -0,0 +1,153 @@
+package keybase
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/keybase/client/go/chat"
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/types"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/kyokomi/emoji"
+	context "golang.org/x/net/context"
+)
+
+type Person struct {
+	KeybaseUsername string
+	KeybaseAvatar   string
+	IsBot           bool
+}
+
+type Message struct {
+	ID        int
+	Kind      string // "Text" | "Reaction"
+	Plaintext string
+	From      *Person
+	At        int64
+}
+
+type ChatNotification struct {
+	// Ordered from oldest to newest
+	Message   *Message
+	ConvID    string
+	TeamName  string
+	TopicName string
+	// e.g. "keybase#general, CoolTeam, Susannah,Jake"
+	ConversationName    string
+	IsGroupConversation bool
+	IsPlaintext         bool
+	SoundName           string
+	BadgeCount          int
+}
+
+func HandleBackgroundNotification(strConvID, body string, intMembersType int, displayPlaintext bool,
+	intMessageID int, pushID string, badgeCount, unixTime int, soundName string, pusher PushNotifier) (err error) {
+	if err := waitForInit(5 * time.Second); err != nil {
+		return nil
+	}
+	gc := globals.NewContext(kbCtx, kbChatCtx)
+	ctx := globals.ChatCtx(context.Background(), gc,
+		keybase1.TLFIdentifyBehavior_CHAT_GUI, nil, chat.NewCachingIdentifyNotifier(gc))
+
+	defer kbCtx.CTrace(ctx, fmt.Sprintf("HandleBackgroundNotification(%s,%v,%d,%d,%s,%d,%d)",
+		strConvID, displayPlaintext, intMembersType, intMessageID, pushID, badgeCount, unixTime),
+		func() error { return flattenError(err) })()
+
+	// Unbox
+	if !kbCtx.ActiveDevice.HaveKeys() {
+		return libkb.LoginRequiredError{}
+	}
+	mp := chat.NewMobilePush(gc)
+	uid := gregor1.UID(kbCtx.Env.GetUID().ToBytes())
+	bConvID, err := hex.DecodeString(strConvID)
+	if err != nil {
+		kbCtx.Log.CDebugf(ctx, "HandleBackgroundNotification: invalid convID: %s msg: %s", strConvID, err)
+		return err
+	}
+	convID := chat1.ConversationID(bConvID)
+	membersType := chat1.ConversationMembersType(intMembersType)
+	msgUnboxed, err := mp.UnboxPushNotification(ctx, uid, convID, membersType, body)
+
+	if err != nil {
+		kbCtx.Log.CDebugf(ctx, "unboxNotification: failed to unbox: %s", err)
+		return err
+	}
+
+	conv, err := utils.GetVerifiedConv(ctx, gc, uid, convID, types.InboxSourceDataSourceAll)
+	if err != nil {
+		kbCtx.Log.CDebugf(ctx, "Failed to get conversation info", err)
+		return err
+	}
+
+	// TODO how to figure out if it's a bot?
```

Comment:
`isBot := gregor1.UITPtrEq(msg.Valid().ClientHeader.BotUID, msg.Valid().ClientHeader.Sender)`

you can put this as a helper in `chat1/extras.go` if it will be useful elsewhere..